src/databricks/labs/ucx/workspace_access/manager.py

```diff
@@ -144,6 +148,51 @@ def apply_group_permissions(self, migration_state: MigrationState) -> bool:
         logger.info("Permissions were applied")
         return True
 
+    def apply_group_permissions_experimental(self, migration_state: MigrationState) -> bool:
+        if len(migration_state) == 0:
+            logger.info("No valid groups selected, nothing to do.")
+            return True
+        logger.info(
+            f"Applying the permissions to account groups. "
+            f"Total groups to apply permissions: {len(migration_state)}."
+        )
+        errors: list[Exception] = []
+        for migrated_group in migration_state.groups:
+            errors.extend(self._migrate_group_permissions_paginated(migrated_group))
+        if len(errors) > 0:
+            logger.error(f"Detected {len(errors)} errors while applying permissions")
+            raise ManyError(errors)
+        logger.info("Permissions were applied")
+        return True
+
+    def _migrate_group_permissions_paginated(self, migrated_group: MigratedGroup) -> list[Exception]:
+        batch_size = 1000
+        errors: list[Exception] = []
+        logger.info(
+            f"Migrating permission from workspace group {migrated_group.name_in_workspace} "
+            f"to account group: {migrated_group.name_in_account}."
+        )
+        while True:
+            try:
+                response = self._ws.permission_migration.migrate_permissions(
+                    self._ws.get_workspace_id(),
+                    migrated_group.name_in_workspace,
+                    migrated_group.name_in_account,
+                    size=batch_size,
+                )
+                # response shouldn't be empty
+                if response.permissions_migrated is None:
+                    break
+                # no more permissions to migrate
+                if response.permissions_migrated == 0:
+                    logger.info("No more permission to migrated.")
+                    break
+                logger.info(f"Migrated {response.permissions_migrated} permissions.")
+            except RuntimeError as e:
+                errors.append(e)
+                break
+        return errors
```

Comment:
```suggestion
```
1. why do we need to return a list of errors, if it has at most one item?
2. won't it be simple to bubble-up the platform error instead?